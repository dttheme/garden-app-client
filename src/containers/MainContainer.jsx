import React from 'react';
import { connect } from 'react-redux';
import Header from './Header';
import Alert from '../components/Alert'
import { userLogout } from '../actions/index.actions'
// import logo from './img/logo.svg';


const Main = props => (
  <div>
    <div>
      <Header logoutUser={props.logout} isLoggedIn={props.user.isLoggedIn} firstName={props.user.firstName} />
        {props.appState.message &&
      <div className="error">
          <Alert
              isError={props.appState.message.isError}
              message={props.appState.message.message}
              isSelfClosing={!props.appState.message.isError}
          />
      </div>
  }
    </div>
      {props.children}
  </div>
)

const mapStateToProps = state => ({
  appState: state.appState,
  user: state.user,
});

const matchDispatchToProps = dispatch => ({
  logout: userLogout.bind(null, dispatch)
})

export default connect(mapStateToProps, matchDispatchToProps)(Main);
