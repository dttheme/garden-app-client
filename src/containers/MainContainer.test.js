import React from 'react';
import { shallow } from 'enzyme';
import { MainContainer } from './MainContainer';

describe('<MainContainer />', () => {

  it('should render page', () => {
    const fakeUser = {
      id: '123',
      username: 'j@e.com',
      firstName: 'Joe',
      isLoggedIn: true,
    }
    const fakeAppState = {
        message: null
    }
    const wrapper = shallow(<MainContainer user={fakeUser} appState={fakeAppState} />);
    expect(wrapper).toMatchSnapshot()
  });

});
