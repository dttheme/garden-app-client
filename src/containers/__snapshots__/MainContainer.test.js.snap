// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContainer /> should render page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    appState={
        Object {
            "message": null,
          }
    }
    user={
        Object {
            "firstName": "Joe",
            "id": "123",
            "isLoggedIn": true,
            "username": "j@e.com",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Header
                    firstName="Joe"
                    isLoggedIn={true}
                    logoutUser={undefined}
          />
</div>,
        <div
          className="vh-100 w-100 overflow-auto fl"
          style={
                    Object {
                              "background": "url(dirt.jpg) no-repeat",
                              "backgroundAttachment": "fixed",
                              "backgroundSize": "cover",
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header
              firstName="Joe"
              isLoggedIn={true}
              logoutUser={undefined}
/>,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "firstName": "Joe",
              "isLoggedIn": true,
              "logoutUser": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "vh-100 w-100 overflow-auto fl",
          "style": Object {
            "background": "url(dirt.jpg) no-repeat",
            "backgroundAttachment": "fixed",
            "backgroundSize": "cover",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Header
                        firstName="Joe"
                        isLoggedIn={true}
                        logoutUser={undefined}
            />
</div>,
          <div
            className="vh-100 w-100 overflow-auto fl"
            style={
                        Object {
                                    "background": "url(dirt.jpg) no-repeat",
                                    "backgroundAttachment": "fixed",
                                    "backgroundSize": "cover",
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Header
                firstName="Joe"
                isLoggedIn={true}
                logoutUser={undefined}
/>,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "firstName": "Joe",
                "isLoggedIn": true,
                "logoutUser": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            null,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "vh-100 w-100 overflow-auto fl",
            "style": Object {
              "background": "url(dirt.jpg) no-repeat",
              "backgroundAttachment": "fixed",
              "backgroundSize": "cover",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
